### PR TITLE
Staging sites: Remove staging banner

### PR DESCRIPTION
**Pull request by matt-west**
Fixes https://github.com/Automattic/dotcom-forge/issues/7948

## Proposed Changes

* Remove the yellow and black banner that's displayed on staging sites.
* Remove the icon next to the `Staging Site` header.
* Clean up the header styles so they match when switching between production and staging sites.

| Before | After |
|--------|--------|
| Screenshot 2024-07-11 at 16 48 31 | Screenshot 2024-07-11 at 16 48 03 |

## Why are these changes being made?

* We now display a yellow border and buttons when viewing a staging site, so we don't need to signal that status from within the `Staging Site` tab.

## Testing Instructions

* Run with `yarn start`
* Create a site
* Navigate to the `Staging Site` tab
* Click the `Add staging site` button
* Switch to the staging site and note that there yellow and black banner is no longer shown on the `Staging Site` tab.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/hosting/staging-site-card/staging-site-production-card.tsx [hunk 1]
```diff
@@ -1,12 +1,10 @@
-import { Button, Card, Gridicon } from '@automattic/components';
+import { Button, Card } from '@automattic/components';
 import styled from '@emotion/styled';
 import { useI18n } from '@wordpress/react-i18n';
 import clsx from 'clsx';
 import { localize } from 'i18n-calypso';
 import { useEffect, useState } from 'react';
 import { connect } from 'react-redux';
-import dividerPattern from 'calypso/assets/images/hosting/divider-pattern.svg';
```
Comment (sejas) [inline client/my-sites/hosting/staging-site-card/staging-site-production-card.tsx:8]: I think this is the only place where we use this `svg` divider. I've searched for `divider-pattern.svg` in the code base and this is the only result I got.

Let's remove the file too.

**Review by sejas — APPROVED** [review]
Thanks for creating the PR! 🙏 

I tested it by creating a staging site and accessing `https://wordpress.com/staging-site/staging-48a4-20240705gbatomic.wpcomstaging.com` in production and also in this branch and I confirm the separator and icon are gone. 

- [x] Let's also remove `client/assets/images/hosting/divider-pattern.svg`

Let me know if you want me to deploy the PR. Happy to help with it.

| Before | After |
|--------|--------|
Screenshot 2024-07-11 at 22 42 50 |  Screenshot 2024-07-11 at 22 42 53 |

**Comment by wojtekn** [comment]
@matt-west, instead of removing, would it make sense to move the banner to the top of the parent block or add another mark showing it is a staging site? The yellow border seems a bit too subtle as the single design element.

File: client/my-sites/hosting/staging-site-card/staging-site-production-card.tsx (hunk 1 above)
Comment (matt-west) [reply client/my-sites/hosting/staging-site-card/staging-site-production-card.tsx:8]: Good call. I've removed that in [104346e](https://github.com/Automattic/wp-calypso/pull/92611/commits/104346eceb08d858802b57191248602cc347d1bd)

**Comment by matt-west** [comment]
> would it make sense to move the banner to the top of the parent block or add another mark showing it is a staging site? The yellow border seems a bit too subtle as the single design element.

I considered this, but it feels like overkill. I think the outer border, the yellow `My Home` button, and the `Staging` label next to the site URL adequately communicate that the user is look at a staging site.

**Pull request merged**